pages/profile.php

```diff
@@ -207,53 +211,56 @@ function userProfile() {
-
-		var stamina = {$player->stamina};
-		var max_stamina = {$player->max_stamina};
+        var statusBars = {
+            health: [{$player->health}, {$player->max_health}, 0],         
+            chakra: [{$player->chakra}, {$player->max_chakra}, 0],
+            stamina: [{$player->stamina}, {$player->max_stamina}, 0],     
+        };
 
 		var regen = {$player->regen_rate} + {$player->regen_boost}; //no regen cut
 
 		setInterval(() => {
 
-			document.getElementById('regentimer').innerHTML = remainingtime; //minus 1 to compensate for lag
+			$('#regentimer').text(remainingtime); //minus 1 to compensate for lag
 
 
 			if(remainingtime <= 0){
 				remainingtime = 60;
 
-				if((health + regen) >= max_health){
-					health = max_health;
-				} else {
-					health += regen; //health ignores regen boost
-				}
-
-				if((chakra + regen) >= max_chakra){
-					chakra = max_chakra;
-				} else {
-					chakra += regen;
-				}
-
-				if((stamina + regen) >= max_stamina){
-					stamina = max_stamina;
-				} else {
-					stamina += regen;
-				}
-
 			//update health amounts / bars
-			document.getElementById('health').innerHTML = health.toFixed(2) + '/' + max_health.toFixed(2);
-			document.getElementById('healthbar').style.width = ( health / max_health )*100 + '%';
-
-			document.getElementById('chakra').innerHTML = chakra.toFixed(2) + '/' + max_chakra.toFixed(2);
-			document.getElementById('chakrabar').style.width = ( chakra / max_chakra )*100 + '%';
+			for (const bar in statusBars)
+			{            
+                if(bar === 'health')
+                {
+                    if(statusBars[bar][0] + (regen * 2) > statusBars[bar][1])
+                    {
+                        statusBars[bar][0] = statusBars[bar][1];
+                        statusBars[bar][2] = statusBars[bar][1];
+                    }
+                    else
+                    {
+                        statusBars[bar][0] += (regen * 2);
+                        statusBars[bar][2] =  statusBars[bar][0] + (regen * 2);
+                    }                                
+                }
+                else if (statusBars[bar][0] + regen > statusBars[bar][1])
+                {
+                    statusBars[bar][0] = statusBars[bar][1];
+                    statusBars[bar][2] = statusBars[bar][1];
+                }
+                else 
+                {
+                    statusBars[bar][0] += regen;
+                    statusBars[bar][2] =  statusBars[bar][0] + regen;
+                }
+                //console.log(bar, statusBars[bar]);
+                if(bar.startsWith('max')) continue;
+                $('#' + bar).html((statusBars[bar][0] == statusBars[bar][1])? statusBars[bar][0].toFixed(2) + '/' + statusBars[bar][1].toFixed(2) : statusBars[bar][0].toFixed(2) + '/' + statusBars[bar][1].toFixed(2) + '-> <b style=\'color: green\'>' + statusBars[bar][2].toFixed(2) + '</b>');
+			    $('#' + bar + 'bar').val(statusBars[bar][0]);  
+            };
```

Comment:
I actually think hardcoding the value setting for each bar is clearer than making an object and iterating through it here - Reading through this loop it's a bit difficult to understand what's going on with sections like

```
statusBars[bar][0] = statusBars[bar][1];
statusBars[bar][2] = statusBars[bar][1];
```

Could this block be refactored to use hardcoded statements for each bar type, so it's easier to read the logic you're trying to do here? I think it'll actually end up being less code and easier to read and see what's going on

Reply:
Sure, I can do that. I turned it into an object to make it easier to set all of the proper values since you are doing multiple actions against the same bit of data.